sys/kern/time.c

```diff
@@ -144,11 +159,143 @@ timespec_t nanotime(void) {
   return tp;
 }
 
+static systime_t tv2hz(const timeval_t *tv) {
+  timespec_t ts;
+  tv2ts(tv, &ts);
+  return ts2hz(&ts);
+}
+
+static timeval_t microuptime(void) {
+  timeval_t now;
+  bintime_t btnow = binuptime();
+  bt2tv(&btnow, &now);
+  return now;
+}
+
+static void kitimer_get(proc_t *p, kitimer_t *timer, struct itimerval *tval) {
+  assert(mtx_owned(&p->p_lock));
+
+  timeval_t now = microuptime();
+
+  kitimer_t *it = &p->p_itimer;
+  if (timercmp(&it->kit_next, &now, <=))
+    timerclear(&tval->it_value);
+  else
+    timersub(&it->kit_next, &now, &tval->it_value);
+
+  tval->it_interval = it->kit_interval;
+}
```

Comment:
Why `timer` is ignored here?

Reply:
The `timer` parameter is redundant, so I have removed it.